addons/website/static/src/js/components/survey/survey.js

```diff
@@ -0,0 +1,807 @@
+                const paletteColor2 = state.paletteSelection.allPalettes[paletteName]['color2'];
+                const delta1 = deltaE(color1, paletteColor1);
+                const delta2 = deltaE(color2, paletteColor2);
+                state.paletteSelection.allPalettes[paletteName]['score'] = (delta1 + delta2) / 2;
+              }
+              palettes = Object.values(state.paletteSelection.allPalettes).sort((a, b) => a['score'] - b['score']);
+              palettes[0].type = 'recommended';
+            } else {
+              palettes = Object.values(state.paletteSelection.allPalettes);
+            }
+            const selectedPalettes = {};
+            for (let i = 0; i < maxPalette; i += 1) {
+              selectedPalettes[palettes[i].id] = palettes[i];
+              if (i > 0) {
+                palettes[i].type = 'base';
+              }
+            }
+            if (selectedPalettes) {
+              state.paletteSelection.palettes = selectedPalettes;
+            }
+          },
+          updateRecommendedThemes({state}, themes) {
+            if (themes[0]) {
+              state.themeSelection.firstTheme.name = themes[0].name;
+              state.themeSelection.firstTheme.url = themes[0].url;
+              state.themeSelection.firstTheme.id = themes[0].id;
+            }
+            if (themes[1]) {
+              state.themeSelection.secondTheme.name = themes[1].name;
+              state.themeSelection.secondTheme.url = themes[1].url;
+              state.themeSelection.secondTheme.id = themes[1].id;
+            }
+            if (themes[2]) {
+              state.themeSelection.thirdTheme.name = themes[2].name;
+              state.themeSelection.thirdTheme.url = themes[2].url;
+              state.themeSelection.thirdTheme.id = themes[2].id;
+            }
+          }
+        };
+
+        const getters = {
+          getWebsiteTypes({state}) {
+            return Object.values(state.description.descriptionData.websiteTypes).map(x => x);
+          },
+
+          getSelectedType({state}, id) {
+            return id ? state.description.descriptionData.websiteTypes[id] : undefined;
+          },
+
+          getWebsitePurpose({state}) {
+            return Object.values(state.description.descriptionData.websitePurposes).map(x => x);
+          },
+
+          getSelectedPurpose({state}, id) {
+            return id ? state.description.descriptionData.websitePurposes[id] : undefined;
+          },
+
+          getFeatures({state}) {
+            const columnNumber = 3;
+            const featureRows = [];
+
+            const features = Object.values(state.featureSelection.features);
+            let currentRow = [];
+            for (let i = 0; i < features.length; i += 1) {
+              currentRow.push(features[i]);
+              if (currentRow.length === columnNumber) {
+                featureRows.push(currentRow);
+                currentRow = [];
+              }
+            }
```

Comment:
If the only thing you're doing with a loop variable when looping over an array is accessing the corresponding array item, prefer `for ... of` or `Array.forEach`. But in this case you can do better with slices:

```js
 for (let i = 0; i < features.length; i += columnNumber) {
  featureRows.push(features.slice(i, i + columnNumber));
}
```

In general, it seems like this function is used for layouting, any reason you can't use flex-wrap and let the browser do that for you instead? If you can't you should probably consolidate the row-building logic into a separate function since you do the same thing for palettes.